Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/Internal/ThreadSafeMongoMobile.swift

```diff
@@ -134,3 +134,29 @@ class ThreadSafeMongoCollection<T: Codable>: Codable {
         return try underlyingCollection().deleteMany(filter, options: options)
     }
 }
+
+extension ThreadSafeMongoCollection where T == Document {
+    @discardableResult
+    func insertOne(_ value: inout T) throws -> InsertOneResult? {
```

Comment:
[q] why did we do this? I know we're the only consumer of this but making these inout seems smelly and the behavior of modifying the argument seems a little odd to me, especially if the Swift driver isn't doing it. is there another way to do this?

Reply:
This is the spec-based behaviour the driver used to have, that it has since removed. We rely heavily on this behaviour. The other way to do this would be to do this inline.